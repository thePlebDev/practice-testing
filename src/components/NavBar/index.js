import React,{useState} from 'react';



const NavBar = ()=>{
  const [state,setState] =useState(false)

  const handleClick =()=>{
    setState(!state)
    console.log(state)
  }

  return(
    <div>
      <h1>The navBar</h1>
      <button className={state?"div-button-open":"div-button-close"} onClick={()=>handleClick()}>Click me</button>
    </div>
  )
}

export default NavBar
