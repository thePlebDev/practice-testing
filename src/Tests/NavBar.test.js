import React from 'react';
import {shallow,mount} from 'enzyme';
import renderer from 'react-test-renderer';

import NavBar from '../components/NavBar';

// the snapshot works
it('renders correctly when there are no items',()=>{
  const nav = renderer.create(<NavBar />).toJSON();
  expect(nav).toMatchSnapshot();
})

checks to make sure that the props are in
it('renders with the inpur correctly with the null value',()=>{
  const props ={
    value:null
  },
  wrapper = mount(<NavBar {...props} />);
  expect(wrapper.prop('value')).toEqual(null)

})

describe('the navBar opening and closing event',()=>{
  // this tested the state and the className change
  it('clicks and changes the className on the element.',()=>{
    const wrapper = shallow(<NavBar />)
    // we know that this works
    expect(wrapper.find('.div-button-close').length).toEqual(1)
    wrapper.find('.div-button-close').simulate('click');
    expect(wrapper.find('.div-button-open').length).toEqual(1)

  })
})

// so right now I want to test some useEffect stuff.
