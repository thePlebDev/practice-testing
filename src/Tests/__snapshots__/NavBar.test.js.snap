// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <h1>
    The navBar
  </h1>
  <button
    className="div-button-close"
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
